aws-kendra-datasource/src/test/java/software/amazon/kendra/datasource/convert/ConfluenceConverterTest.java

```diff
@@ -0,0 +1,78 @@
+package software.amazon.kendra.datasource.convert;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import software.amazon.kendra.datasource.DataSourceConfiguration;
+import software.amazon.kendra.datasource.convert.confluence.ConfluenceConverter;
+
+public class ConfluenceConverterTest {
+
+    @ParameterizedTest
+    @MethodSource("provideTestCases")
+    public void testConversionWorksAsExpected(ConfluenceConverterTestCase testCase) {
+
+        if (testCase.getInput() instanceof ConfluenceConverterTestCase.Model) {
+            // model -> sdk
+            assertThat(ConfluenceConverter.toSdkDataSourceConfiguration(testCase.getInput().asModel()))
+                .as(String.format("model -> sdk : %s", testCase.getDescription()))
+                .isEqualTo(testCase.getOutput().asSdk());
+
+        } else {
```

Comment:
I think it'd be a bit cleaner to separate these out into two tests so we don't have the if statement. Not a blocker though.